commit 022f5af61bc3cbfa2b74ab355b0d2d30b3dab027
Author: Khem Raj <raj.khem@gmail.com>
Date:   Sat Dec 22 15:28:40 2018 -0800

    tests/amd64: Do not clobber %rsp register
    
    This is seen with gcc-9.0 compiler now which is fix that gcc community
    did recently
    https://gcc.gnu.org/bugzilla/show_bug.cgi?id=52813
    
    Signed-off-by: Khem Raj <raj.khem@gmail.com>

diff --git a/none/tests/amd64-linux/bug345887.c b/none/tests/amd64-linux/bug345887.c
index 0f9237d..269bd70 100644
--- a/none/tests/amd64-linux/bug345887.c
+++ b/none/tests/amd64-linux/bug345887.c
@@ -20,13 +20,17 @@ static void inner(void)
       "movq $0x10d, %%r14\n"
       "movq $0x10e, %%r15\n"
       // not %rbp as mdb is then not able to reconstruct stack trace
+      // Do change %rsp (to test a bogus stack pointer),
+      // but don't add %rsp to the clobber list since gcc ignores it
+      // and since gcc >= 9.0 errors about it
+      // see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=52813
       "movq $0x10f, %%rsp\n"
       "movq $0x1234, (%%rax)\n"  // should cause SEGV here
       "ud2"                      // should never get here
       : // no output registers
       : // no input registers
       : "memory", "%rax", "%rbx", "%rcx", "%rdx", "%rsi", "%rdi",
-        "%r8", "%r9", "%r10", "%r11", "%r12", "%r13", "%r14", "%r15", "%rsp");
+        "%r8", "%r9", "%r10", "%r11", "%r12", "%r13", "%r14", "%r15");
 }
 
 __attribute__((noinline))
commit c512949082c4fc2285a82e102d4212c66e034a31
Author: Mark Wielaard <mark@klomp.org>
Date:   Fri Jan 11 20:00:17 2019 +0100

    Bug 402480 Do not use %esp in clobber list.
    
    This is the same fix as for amd64-linux, but now for x86-linux.

diff --git a/none/tests/x86-linux/bug345887.c b/none/tests/x86-linux/bug345887.c
index f2e8a8d..ef21d4e 100644
--- a/none/tests/x86-linux/bug345887.c
+++ b/none/tests/x86-linux/bug345887.c
@@ -12,12 +12,14 @@ static void inner(void)
       "movl $0x105, %%esi\n"
       "movl $0x106, %%edi\n"
       // not %ebp as mdb is then not able to reconstruct stack trace
+      // clobbering %esp is really bad, but that is kind of the point
+      // we don't add it to the clobber list since gcc9 warns about that
       "movl $0x108, %%esp\n"
       "movl $0x1234, (%%eax)\n"  // should cause SEGV here
       "ud2"                      // should never get here
       : // no output registers
       : // no input registers
-      : "memory", "%eax", "%ebx", "%ecx", "%edx", "%esi", "%edi", "%esp");
+      : "memory", "%eax", "%ebx", "%ecx", "%edx", "%esi", "%edi");
 }
 
 __attribute__((noinline))
